.github/workflows/main.yml

```diff
@@ -35,21 +37,41 @@ jobs:
     name: test code
     runs-on: ubuntu-latest
     timeout-minutes: 3
+    strategy:
+      matrix:
+        dbt-core:
+          - version: "1.2.0-latest"
+            package: "dbt-core~=1.2.0"
+            server-branch: "0.1.latest"
+          - version: "1.3.0-latest"
+            package: "dbt-core~=1.3.0"
+            server-branch: "0.1.latest"
+          - version: "1.4.0-latest"
+            package: "dbt-core~=1.4.0"
+            server-branch: "0.1.latest"
+          # - version: "1.5.0-pre"
+          #   package: "dbt-core~=1.5.0b1"
+          #   server-branch: "main"
+          - version: "head"
+            package: https://github.com/dbt-labs/dbt-core/archive/HEAD.tar.gz#egg=dbt-core&subdirectory=core
+            server-branch: "main"
     steps:
       - name: checkout repo
+        if: ${{ matrix.dbt-core.server-branch == github.ref_name || matrix.dbt-core.server-branch == github.event.pull_request.base.ref }}
```

Comment:
I noticed this condition and first want to learn from you it's meaning. matrix.dbt-core.server-branch == github.ref_name is for push action while github.event.pull_request.base.ref is for pull request action.

Secondly can we use job level if instead of step level if? So we don't need to duplicate condition three times.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif

Reply:
The conditional just means to only run if it's on the `server-branch` or if it's a PR to that `server-branch` -- so yes, the first will only evaluate to True on a push and the second will evaluate to True on a PR. 

To your second point-- yes! I didn't realize job-level was possible, thank you for finding that 👍

Reply:
Oh! I'm wrong on the first point, I know why this isn't job level -- this conditional won't have access to the matrix and those values until it starts looping through them

Reply:
Thanks Rachel! I found this https://github.com/actions/runner/issues/1985 about matrix context if job level if, seems like lot's of folks complained about it...

While I'm spending some times digging into this and I maybe find another workaround using env for better readability. We can define env under job, like IS_INTERESTED_SERVER_BRANCH those envs can be reused below so we sync the logic between three steps.

Let me know your thoughts on this! Do you think it would be better from readability perspective? My feeling is that it's easier to understand. I don't have strong opinion on this, just because I don't like repeated code pieces. 

strategy:
      matrix:
        color: ["red", "pink"]
    env: 
      COLOR_ENV: ${{ matrix.color == 'pink' }}
    steps:
      - if: ${{ env.COLOR_ENV == 'true' }}
See run results in https://github.com/dichenqiandbt/test-dbt.

Reply:
I don't think I quite understand how the env value is being set in your example-- are you defining that in the repo settings? I can see in the actions how it's conditionally running actions based on that env value, but where can I see what the conditional actually is? I have no problem DRYing up the code as long as it's still easy to tell what we're building with each branch!

It's also an option for this actions file to be different for different branches, and I could leave off all of the conditionals. I thought it might be better to be explicit, but I don't have strong feelings either 🙂

Reply:
@peter-bertuglia Curious for your thoughts here as well-- these actions are a bit messy, I'm down for whatever is clearest!